oracle/feed_stork.go

```diff
@@ -0,0 +1,297 @@
+	if err := toml.Unmarshal(body, &config); err != nil {
+		err = errors.Wrap(err, "failed to unmarshal TOML config")
+		return nil, err
+	}
+
+	return &config, nil
+}
+
+// NewStorkPriceFeed returns price puller
+func NewStorkPriceFeed(cfg *StorkFeedConfig) (PricePuller, error) {
+	pullInterval := 1 * time.Minute
+	if len(cfg.PullInterval) > 0 {
+		interval, err := time.ParseDuration(cfg.PullInterval)
+		if err != nil {
+			err = errors.Wrapf(err, "failed to parse pull interval: %s (expected format: 60s)", cfg.PullInterval)
+			return nil, err
+		}
+
+		if interval < 1*time.Second {
+			err = errors.Wrapf(err, "failed to parse pull interval: %s (minimum interval = 1s)", cfg.PullInterval)
+			return nil, err
+		}
+
+		pullInterval = interval
+	}
+
+	var oracleType oracletypes.OracleType
+	if cfg.OracleType == "" {
+		oracleType = oracletypes.OracleType_PriceFeed
+	} else {
+		tmpType, exist := oracletypes.OracleType_value[cfg.OracleType]
+		if !exist {
+			return nil, fmt.Errorf("oracle type does not exist: %s", cfg.OracleType)
+		}
+
+		oracleType = oracletypes.OracleType(tmpType)
+	}
+
+	feed := &storkPriceFeed{
+		ticker:       cfg.Ticker,
+		providerName: cfg.ProviderName,
+		interval:     pullInterval,
+		url:          cfg.Url,
+		header:       cfg.Header,
+		message:      cfg.Message,
+		oracleType:   oracleType,
+
+		logger: log.WithFields(log.Fields{
+			"svc":      "oracle",
+			"dynamic":  true,
+			"provider": cfg.ProviderName,
+		}),
+
+		svcTags: metrics.Tags{
+			"provider": cfg.ProviderName,
+		},
+	}
+
+	return feed, nil
+}
+
+func (f *storkPriceFeed) Interval() time.Duration {
+	return f.interval
+}
+
+func (f *storkPriceFeed) Symbol() string {
+	return f.ticker
+}
+
+func (f *storkPriceFeed) Provider() FeedProvider {
+	return FeedProviderStork
+}
+
+func (f *storkPriceFeed) ProviderName() string {
+	return f.providerName
+}
+
+func (f *storkPriceFeed) OracleType() oracletypes.OracleType {
+	return oracletypes.OracleType_Stork
+}
+
+// PullAssetPair pulls asset pair for an asset id
+func (f *storkPriceFeed) PullAssetPair(ctx context.Context) (assetPairs oracletypes.AssetPair, err error) {
+	metrics.ReportFuncCall(f.svcTags)
+	doneFn := metrics.ReportFuncTiming(f.svcTags)
+	defer doneFn()
+
+	// Parse the URL
+	u, err := url.Parse(f.url)
+	if err != nil {
+		log.Fatal("Error parsing URL:", err)
+		return oracletypes.AssetPair{}, nil
+	}
+	header := http.Header{}
+	header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(f.header)))
+
+	dialer := websocket.DefaultDialer
+	dialer.EnableCompression = true
+
+	// Connect to the WebSocket server
+	conn, resp, err := dialer.Dial(u.String(), header)
+	if err != nil {
+		if resp != nil {
+			log.Printf("Handshake failed with status: %d\n", resp.StatusCode)
+			for k, v := range resp.Header {
+				log.Printf("%s: %v\n", k, v)
+			}
+		}
+		log.Fatal("Error connecting to WebSocket:", err)
+		return oracletypes.AssetPair{}, nil
+	}
+	defer conn.Close()
```

Comment:
we shouldn't initiate and close a websocket connection every time we want to pull price.

Reply:
Moved it to [here](https://github.com/decentrio/injective-price-oracle/blob/4136531e0e7f7627739b6f45bfba0d7ea82152b4/cmd/injective-price-oracle/oracle.go#L259), so we don't need to make the connection every time we want to pull price